File: frontend/src/api/correction.ts
import apiClient from './index'

export interface CorrectionStep {
  step_no: number
  content: string
  is_correct: boolean
  error_type?: string
  hint?: string
}

export interface CorrectionResult {
  correction_id: string
  question_id: string
  steps: CorrectionStep[]
  overall_feedback: string
}

export function analyzeCorrection(file: File, questionId?: string) {
  const formData = new FormData()
  formData.append('image', file)
  if (questionId) formData.append('question_id', questionId)
  return apiClient.post<CorrectionResult>('/correction/analyze', formData, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

File: backend/app/schemas/correction.py
"""Correction schemas."""
from pydantic import BaseModel
from typing import Optional


class CorrectionRequest(BaseModel):
    """批改请求."""
    image_url: str
    question_id: Optional[str] = None


class CorrectionStep(BaseModel):
    """批改步骤."""
    step_no: int
    content: str
    is_correct: bool
    error_type: Optional[str] = None
    hint: Optional[str] = None


class CorrectionAnalyzeResponse(BaseModel):
    """手写批改分析结果."""
    correction_id: str
    question_id: str
    steps: list[CorrectionStep]
    overall_feedback: str
